node/overridden_manager.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package node
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ validators.Manager = (*overriddenManager)(nil)
+
+// newOverriddenManager returns a Manager that overrides of all calls to the
+// underlying Manager to only operate on the validators in [subnetID].
+func newOverriddenManager(subnetID ids.ID, manager validators.Manager) validators.Manager {
```

Comment:
1. Can we just get rid of this since we can just use the struct directly?
2. If we do decide to keep this, can we just return `overriddenManager` instead of the interface? We frequently do this in our code but I think it's better to return the concrete type in almost every case so the caller can decide if it wants to use the interface signature or the concrete type.

Reply:
I prefer to have “new” functions rather than struct initializing. I feel like they’re less error prone in case we expand structs. But I agree returning structs directly (in same package/unexported)